modules/subchannel/src/problems/SubChannel1PhaseProblem.C

```diff
@@ -363,6 +375,58 @@ SubChannel1PhaseProblem::computeInterpolatedValue(PetscScalar topValue,
   return alpha * botValue + (1.0 - alpha) * topValue;
 }
 
+Real
+SubChannel1PhaseProblem::computeNusseltNumber(NusseltStruct nusselt_args)
+{
+  const auto htc_correlation = nusselt_args.htc_correlation;
+  const auto Re = nusselt_args.Re;
+  const auto Pr = nusselt_args.Pr;
+
+  switch (htc_correlation)
+  {
+    case 0: // dittus-boelter
+      return 0.023 * std::pow(Re, 0.8) * std::pow(Pr, 0.4);
+
```

Comment:
thoughts about checking the ranges?

Reply:
Yes, we should. Warning if out of range good?

Reply:
https://mooseframework.inl.gov/source/interfaces/SolutionInvalidInterface.html
this is probably the best option here
the macro is flagInvalidSolution(some message)
and there s code to invalidate the solution

BUT
might not work for SCM, we have not tried
if not, do mooseDoOnce(mooseWarning(...)); warnings can get out of hand

Reply:
Thanks! Seems that SolutionInvalidInterface is working for subchannel. Also, extended the correlation regimes to all Re number (allowing laminar numbers and transition regimes). In this way, we only check for the Pr number. One issue is that Dittus Boelter is not really valid below Pr<0.7. So, almost all the liquid metal tests should fail. May need some significant regolding